ale_linters/erlang/erlc.vim

```diff
@@ -17,7 +17,7 @@ function! ale_linters#erlang#erlc#Handle(buffer, lines) abort
     " error.erl:4: variable 'B' is unbound
     " error.erl:3: Warning: function main/0 is unused
     " error.erl:4: Warning: variable 'A' is unused
-    let l:pattern = '\v^([^:]+):(\d+): (Warning: )?(.+)$'
+    let l:pattern = '\v\C:(\d+): (Warning: )?(.+)$'
```

Comment:
I wonder, can `erlc` return the names of other included files?

Reply:
Indeed it is possible in theory.
Due to how erlc is used in the checker with a temp file, this is unlikely (and causes unwanted errors, such as printing warnings about undefined behaviors when using user defined behaviors).

We could modify the pattern to enable extracting the file name, for instance starting the pattern with ` \v^([a-zA-Z]?:?[^:]+)` instead of `\v^([^:]+)`.
This way we keep the ability to use the file name in the future.